### ISSUE TITLE
Using Controller with React DatePicker

### ISSUE THREAD

**Issue by iwarner** [react-hook-form/react-hook-form#1242]
I am trying to get some sanitised error reporting on the DatePicker custom input

**Codesandbox link (Required)**

Please check this out
https://codesandbox.io/s/react-hook-form-and-react-date-picker-wrapped-at-controller-cyc48

- When first loaded the errors are {} and display the error style
- If I hit submit immediately DatePicker takes over and does some HTML5 validation which I dont want
- Selecting a date and hitting submit does not remove the errors as still {}

Basically the issue is about getting the errors to validate correctly within this Controller situation

Many thanks.

**Comment by bluebill1049** [comment]
please wrap your input into a proper form.

```
<Controller
            as={
              <ReactDatePicker
                dateFormat="d MMM yyyy"
                minDate={new Date()}
                showTimeSelect={false}
                todayButton="Today"
                customInput={<StyledInput errors={errors} />}
                dropdownMode="select"
                isClearable
                placeholderText="Click to select time"
                shouldCloseOnSelect
              />
            }
            control={control}
            register={register({ required: true })}
            name="expiryAt"
            valueName="selected" // DateSelect value's name is selected 
            onChange={([selected]) => {
              return selected;
            }}
/>
```

update your CSB and we can go from there. Keep your CSB simple as well. it helps me to find the issue.

**Comment by iwarner** [comment]
Sure thing
I copied yours from here though:
https://codesandbox.io/s/react-hook-form-and-react-nice-dates-wrapped-at-controller-qogy4

Will update now

Considered spectrum but wasnt sure this error thing was a bug or not.

**Comment by bluebill1049** [comment]
> Considered spectrum but wasnt sure this error thing was a bug or not.

all good, let's sorts this out together.

**Comment by iwarner** [comment]
Thanks
https://codesandbox.io/s/react-hook-form-and-react-date-picker-wrapped-at-controller-cyc48

Updated

**Comment by bluebill1049** [comment]
> I copied yours from here though:
> https://codesandbox.io/s/react-hook-form-and-react-nice-dates-wrapped-at-controller-qogy4

I copied from someone else and was trying to help as well haha.

**Comment by iwarner** [comment]
Yeah I changed from the nice date pickers to the react-datepicker - I think having these examples in the docs that link to CodePen would be awesome, we are working on a massive form heavy site so happy to help with that, React Select etc.....

**Comment by bluebill1049** [comment]
https://codesandbox.io/s/react-hook-form-and-react-date-picker-wrapped-at-controller-2zz2z

**Comment by bluebill1049** [comment]
> Yeah I changed from the nice date pickers to the react-datepicker - I think having these examples in the docs that link to CodePen would be awesome, we are working on a massive form heavy site so happy to help with that, React Select etc.....

take a look the doc under controller.

**Comment by iwarner** [comment]
Ok testing now - yeah I read all those :)
https://react-hook-form.com/api#Controller
I see the valueName prop now 

will re-read.

If you put in the docs the DatePicker example that needs the valueName that could help others

Many thanks - onto React Select now :)

**Comment by bluebill1049** [comment]
> If you put in the docs the DatePicker example that needs the valueName that could help others

will do :)

**Comment by iwarner** [comment]
Thanks Bill - added you to my sponsors list.. your project is amazing

**Comment by bluebill1049** [comment]
oh wow. Thank you very much! very kind of you ❤️

**Comment by iwarner** [comment]
Its pure bribery as I am sure I will have other queries :) Our main other third party libraires are React-Dropzone and React-Select so will see how that goes, I will try and make some CodeSandbox for these also and then can use in the docs.

**Comment by bluebill1049** [comment]
Awesome, i will update with more examples to that codesandbox to help others.

**Comment by bluebill1049** [comment]
Actually let's pin this issue, so more users can supply more examples when using with Controller.

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/issues/1247

let's track it on this ticket.

**Comment by lacykerel** [comment]
I'm trying this approach and I seem to can't get the value to show up from the datepicker 

`          <Controller
                          as={
                            <DatePicker
                              className="form-control"
                              minDate={new Date()}
                              endDate={endDate}
                              selectsStart
                              showTimeSelect
                              timeFormat="h:mm aa"
                              timeIntervals={30}
                              timeCaption="Time"
                              dateFormat="MM/d, yyyy h:mm aa"
                              placeholderText="Select a start date"
                            />
                          }
                          control={control}
                          register={register({ required: true })}
                          name="start"
                          onChange={([selected]) => {
                            return selected;
                          }}
                          rules={{
                            required: true,
                          }}
                        />`

**Comment by bluebill1049** [comment]
@lacykerel Take a look at the doc: https://react-hook-form.com/api#Controller there is a link for example.

**Comment by lacykerel** [comment]
This helped! Thanks!

**Comment by Rahul-icoder** [comment]
How i can use material ui date picker and time picker in react hook form fieldArray

**Comment by bluebill1049** [comment]
> How i can use material ui date picker and time picker in react hook form fieldArray

should be the same right, as long as you supplied a `defaultValue`.

**Comment by JavaJamie** [comment]
Can you use react-datepicker on the react-hook-form with a customInput? It doesn't work for me
